### ISSUE TITLE
mapping does not work when parameter name is 'size'

### ISSUE THREAD

**Issue by grumpy-programmer** [mapstruct/mapstruct#1244]
Related to #909
mapstruct 1.2.0.Beta3
lombok 1.16.14

Example code:
```
import lombok.Data;
import lombok.Generated;

@Mapper(componentModel = "spring")
public interface SizeMapper {

    @Mapping(source = "width", target = "width")
    @Mapping(source = "height", target = "height")
    SizeDto toSizeDto(Size size);
}

@Data
@Generated
public class SizeDto {

    private Integer width;

    private Integer height;
}

@Data
@Generated
public class Size {

    private Integer width;

    private Integer height;
}
```
Compilation exception:

SizeMapper.java:[12,8] Internal error in the mapping processor: java.lang.RuntimeException: org.mapstruct.ap.shaded.freemarker.template.TemplateModelException: Getting the number of items or enumerating the items is not supported on this method+sequence (wrapper: f.e.bded.freemarker.ext.beans.SimpleMethodModel) value.  (Hint 1: Maybe you wanted to call this method first and then do something with its return value. Hint 2: Getting items by intex possibly works, hence it's a "+sequence".)    ----  FTL stack trace ("~" means nesting-related):  	- Failed at: #list propertyMappingsByParameter[sou...  [in template "org/mapstruct/ap/internal/model/BeanMappingMethod.ftl" at line 65, column 9]  ----  	at org.mapstruct.ap.internal.writer.ModelWriter.writeModel(ModelWriter.java:93)  	at org.mapstruct.ap.internal.processor.MapperRenderingProcessor.createSourceFile(MapperRenderingProcessor.java:74)  at org.mapstruct.ap.internal.processor.MapperRenderingProcessor.writeToSourceFile(MapperRenderingProcessor.java:52)  	at org.mapstruct.ap.internal.processor.MapperRenderingProcessor.process(MapperRenderingProcessor.java:42)  	at org.mapstruct.ap.internal.processor.MapperRenderingProcessor.process(MapperRenderingProcessor.java:37)  	at org.mapstruct.ap.MappingProcessor.process(MappingProcessor.java:280)  	at org.mapstruct.ap.MappingProcessor.processMapperTypeElement(MappingProcessor.java:260)  	at org.mapstruct.ap.MappingProcessor.processMapperElements(MappingProcessor.java:226)  	at org.mapstruct.ap.MappingProcessor.process(MappingProcessor.java:158)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.callProcessor(JavacProcessingEnvironment.java:794)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.discoverAndRunProcs(JavacProcessingEnvironment.java:705)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.access$1800(JavacProcessingEnvironment.java:91)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment$Round.run(JavacProcessingEnvironment.java:1035)  	at com.sun.tools.javac.processing.JavacProcessingEnvironment.doProcessing(JavacProcessingEnvironment.java:1176)  	at com.sun.tools.javac.main.JavaCompiler.processAnnotations(JavaCompiler.java:1170)  	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:856)  	at com.sun.tools.javac.main.Main.compile(Main.java:523)  	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)  	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)  	at org.codehaus.plexus.compiler.javac.JavaxToolsCompiler.compileInProcess(JavaxToolsCompiler.java:126)  	at org.codehaus.plexus.compiler.javac.JavacCompiler.performCompile(JavacCompiler.java:174)  	at org.apache.maven.plugin.compiler.AbstractCompilerMojo.execute(AbstractCompilerMojo.java:943)  	at org.apache.maven.plugin.compiler.CompilerMojo.execute(CompilerMojo.java:137)  	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)  	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)  	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:154)  	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:146)  	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)  	at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)  	at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)  	at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)  	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:309)  	at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:194)  	at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:107)  	at org.apache.maven.cli.MavenCli.execute(MavenCli.java:993)  	at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:345)  	at org.apache.maven.cli.MavenCli.main(MavenCli.java:191)  	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)  	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)  	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)  	at java.lang.reflect.Method.invoke(Method.java:498)  	at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)  	at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)  	at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)  	at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)  Caused by: org.mapstruct.ap.shaded.freemarker.template.TemplateModelException: Getting the number of items or enumerating the items is not supported on this method+sequence (wrapper: f.e.bded.freemarker.ext.beans.SimpleMethodModel) value.  (Hint 1: Maybe you wanted to call this method first and then do something with its return value. Hint 2: Getting items by intex possibly works, hence it's a "+sequence".)    ----  FTL stack trace ("~" means nesting-related):  	- Failed at: #list propertyMappingsByParameter[sou...  [in template "org/mapstruct/ap/internal/model/BeanMappingMethod.ftl" at line 65, column 9]  ----  	at org.mapstruct.ap.shaded.freemarker.ext.beans.SimpleMethodModel.size(SimpleMethodModel.java:93)  	at org.mapstruct.ap.shaded.freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:152)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visitIteratorBlock(Environment.java:501)  	at org.mapstruct.ap.shaded.freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visit(Environment.java:257)  	at org.mapstruct.ap.shaded.freemarker.core.MixedContent.accept(MixedContent.java:57)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visitByHiddingParent(Environment.java:278)  	at org.mapstruct.ap.shaded.freemarker.core.IfBlock.accept(IfBlock.java:48)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visit(Environment.java:257)  	at org.mapstruct.ap.shaded.freemarker.core.MixedContent.accept(MixedContent.java:57)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visit(Environment.java:257)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.process(Environment.java:235)  	at org.mapstruct.ap.shaded.freemarker.template.Template.process(Template.java:262)  	at org.mapstruct.ap.internal.writer.FreeMarkerModelElementWriter.write(FreeMarkerModelElementWriter.java:48)  	at org.mapstruct.ap.internal.writer.FreeMarkerWritable.write(FreeMarkerWritable.java:32)  	at org.mapstruct.ap.internal.writer.ModelIncludeDirective.execute(ModelIncludeDirective.java:61)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visit(Environment.java:333)  	at org.mapstruct.ap.shaded.freemarker.core.UnifiedCall.accept(UnifiedCall.java:100)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visit(Environment.java:257)  	at org.mapstruct.ap.shaded.freemarker.core.MixedContent.accept(MixedContent.java:57)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visit(Environment.java:257)  	at org.mapstruct.ap.shaded.freemarker.core.IteratorBlock$Context.runLoop(IteratorBlock.java:145)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visitIteratorBlock(Environment.java:501)  	at org.mapstruct.ap.shaded.freemarker.core.IteratorBlock.accept(IteratorBlock.java:67)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visit(Environment.java:257)  	at org.mapstruct.ap.shaded.freemarker.core.MixedContent.accept(MixedContent.java:57)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.visit(Environment.java:257)  	at org.mapstruct.ap.shaded.freemarker.core.Environment.process(Environment.java:235)  	at org.mapstruct.ap.shaded.freemarker.template.Template.process(Template.java:262)  	at org.mapstruct.ap.internal.writer.FreeMarkerModelElementWriter.write(FreeMarkerModelElementWriter.java:48)  	at org.mapstruct.ap.internal.writer.FreeMarkerWritable.write(FreeMarkerWritable.java:32)  	at org.mapstruct.ap.internal.writer.ModelWriter.writeModel(ModelWriter.java:84)  	... 44 more

**Comment by filiphr** [comment]
Thanks for reporting this @grumpy-programmer. It is the same issue as in #909. The fix is simple and will be part of the `1.2.0.CR1` release which should come out pretty soon.

The workaround for this is to not use `size` as a parameter for the time being.

Btw. nice name 😄

**Comment by agudian** [comment]
Thanks for reporting @grumpy-programmer, and thanks for the prompt fix, @filiphr!
